dvc/main.py

```diff
@@ -92,11 +101,3 @@ def main(argv=None):
         # Remove cached repos in the end of the call, these are anonymous
         # so won't be reused by any other subsequent run anyway.
         clean_repos()
```

Comment:
Ideally these should be put into atexit kinda like https://github.com/python/cpython/blob/e1945307d36849f8be8937144cf3dd6ebab6274c/Lib/multiprocessing/util.py#L362 . Since they are not only relevant for CLI but for API as well. This also might be the reason why we are getting some weird fini errors in tests.